device/mellanox/x86_64-mlnx_msn2010-r0/plugins/psuutil.py

```diff
@@ -65,14 +66,4 @@ def get_psu_presence(self, index):
         :param index: An integer, 1-based index of the PSU of which to query status
         :return: Boolean, True if PSU is plugged, False if not
         """
-        if index is None:
-            return False
-
-        status = 0
-        try:
-            with open(self.psu_path + self.psu_presence.format(index), 'r') as presence_status:
-                status = int(presence_status.read())
-        except IOError:
-            return False
-
-        return status == 1
+        return isinstance(index, int) and index > 0 and index <= self.get_num_psus()
```

Comment:
@stephenxs IMHO, this is confusing: the code does not provide a status about physical PSU presence.

Reply:
Fixed.
BTW since SN2100 and SN2010 share the codes, I updated SN2100 as well.